Compiler/script/cs_parser.cpp

```diff
@@ -1234,23 +1250,9 @@ int AGS::Parser::ParseParamlist_ParamType(AGS::Symbol param_vartype, bool &param
         cc_error("A function parameter must not have the type 'void'");
         return -1;
     }
-    if (param_is_ptr)
-    {
-        if (!FlagIsSet(_sym.get_flags(param_vartype), kSFLG_Managed))
-        {
-            // can only point to managed structs
-            cc_error("Cannot declare pointer to non-managed type");
-            return -1;
-        }
-        if (param_is_natural_ptr && param_is_autoptr)
-        {
-            cc_error("Cannot use '*' with %s", _sym.get_vartype_name_string(param_vartype).c_str());
-            return -1;
-        }
-    }
     if (FlagIsSet(_sym.get_flags(param_vartype), kSFLG_StructType) && (!param_is_ptr))
     {
-        cc_error("A struct cannot be passed as parameter");
+        cc_error("A non-managed struct cannot be passed as parameter");
```

Comment:
I think the original error message was right here. You pass in a pointer to a managed struct. You don't pass in the struct.

Reply:
Thanks for the hint; you're right, of course.